### PR TITLE
Add notes to debug e2e tests

### PR DESCRIPTION
**Pull request by baijum**
### Motivation

Adding some notes would be helpful for developers.

### Changes

new docs in markdown format.

### Testing

NA

### CODE REVIEW

File: docs/ci.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+The YAML Lint tools validate all the YAML configuration files.  It
+excludes the `vendor` directory while running.  There is a
+configuration file at the top-level directory of the source code:
+`.yamllint`.
+
+### unit
+
+The `unit` target runs the unit tests.  Some of the tests make use of mock
+objects. The unit tests don't require a dedicated OpenShift cluster, unlike
+end-to-end tests.
```
Comment (Avni-Sharma) [inline docs/ci.md:108]: We can add `It runs test-unit` Makefile target. (for example - mentioned in e2e below)

Comment (baijum) [reply docs/ci.md]: Done. 048a77418d81a5799005c7203a4e344bb8cb7d24

**Comment by DhritiShikhar** [comment]
@baijum Is this PR ready or any more changes you will add?

**Comment by baijum** [comment]
> @baijum Is this PR ready or any more changes you will add?

There is still scope for improvement especially the last section. But can we do it later in another PR?

**Comment by DhritiShikhar** [comment]
@baijum Yes. Can you create a GH issue for the same including improvement suggestions?

**Comment by DhritiShikhar** [comment]
@baijum Also, is there a JIRA task for this issue? 
If not, can we add one?

**Pull request merged**